File: 0035/solution.ts
function searchInsert(nums: number[], target: number): number {
  let res = nums.length;
  let l = 0,
    r = nums.length - 1;
  while (l <= r) {
    const p = Math.floor((l + r) / 2);
    if (nums[p] >= target) {
      res = p;
      r = p - 1;
    } else {
      l = p + 1;
    }
  }
  return res;
}

File: 0035/solution.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int l = 0;
        int r = nums.length - 1;
        int res = r + 1;
        while (l <= r) {
            int p = l + (int) Math.floor((r - l) / 2);
            if (target <= nums[p]) {
                res = p;
                r = p - 1;
            } else {
                l = p + 1;
            }
        }
        return res;
    }
}